lib/rack/cache/meta_store.rb

```diff
@@ -64,8 +64,7 @@ def store(request, response, entity_store)
         end
         response.headers['X-Content-Digest'] = digest
         response.headers['Content-Length'] = size.to_s unless response.headers['Transfer-Encoding']
-        # If the entitystore backend is a Noop, do not try to read the body from the backend, it always returns an empty array
-        response.body = entity_store.open(digest) || response.body unless entity_store.is_a? Rack::Cache::EntityStore::Noop
+        response.body = entity_store.open(digest) || response.body
```

Comment:
As a side note, I don't really understand why ```MetaStore#store``` is   reading the body from the cache again, after writing it to the entity store, and overwritting the response body with the body just read from the cache. 

As far as I can tell, this line is pointless and a waste of processing time for most backends, and in the case of ```Noop``` it must not be executed. I'm tempted to suggest removing it altogether but I'm not sure if I'm missing something here.